File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/CurrentProgramModel.java
package com.example.simplehabitassignment.data.vos.models;


import com.example.simplehabitassignment.data.vos.CurrentProgramVO;

public interface CurrentProgramModel {

    void getCurrentProgram(String accessToken, int page, CurrentProgramDelegate delegate);


    interface  CurrentProgramDelegate
    {
        void onSuccess(CurrentProgramVO currentProgram);
        void onFail(String errorMessage);

    }
}

File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/CurrentProgramModel.kt
package com.example.simplehabitassignment.data.vos.models


import com.example.simplehabitassignment.data.vos.CurrentProgramVO

interface CurrentProgramModel {

    fun getCurrentProgram(accessToken: String, page: Int, delegate: CurrentProgramDelegate)


    interface CurrentProgramDelegate {
        fun onSuccess(currentProgram: CurrentProgramVO)
        fun onFail(errorMessage: String)

    }
}
